lib/java/src/org/apache/thrift/transport/TInvalidFrameSizeException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.thrift.transport;
+
+public class TInvalidFrameSizeException extends TTransportException {
```

Comment:
This would be simpler as a TTransportException(TTransportExceptionType::CORRUPTED_DATA) (from C++).  That's the convention currently in the code, to use an exception type, not to derive.  I agree with you in principle as it is easier to unit test changes with specific exception types, just want to point out if doesn't follow the project's currently established conventions.  

At the very least, the TInvalidFrameSizeException should initialize super with that exception type code.